src/passes/OptimizeInstructions.cpp

```diff
@@ -1148,17 +1148,20 @@ struct OptimizeInstructions : public WalkerPass<PostWalker<OptimizeInstructions,
         }
       }
     }
-    // note that this is correct even on floats with a NaN on the left,
-    // as a NaN would skip the computation and just return the NaN,
-    // and that is precisely what we do here. but, the same with -1
-    // (change to a negation) would be incorrect for that reason.
-    if (right->value == Literal::makeFromInt32(1, type)) {
-      if (binary->op == Abstract::getBinary(type, Abstract::Mul) ||
-          binary->op == Abstract::getBinary(type, Abstract::DivS) ||
-          binary->op == Abstract::getBinary(type, Abstract::DivU)) {
-        return binary->left;
+    if (isIntegerType(type) || isFloatType(type)) {
+      // note that this is correct even on floats with a NaN on the left,
+      // as a NaN would skip the computation and just return the NaN,
+      // and that is precisely what we do here. but, the same with -1
+      // (change to a negation) would be incorrect for that reason.
+      if (right->value == Literal::makeFromInt32(1, type)) {
+        if (binary->op == Abstract::getBinary(type, Abstract::Mul) ||
+            binary->op == Abstract::getBinary(type, Abstract::DivS) ||
+            binary->op == Abstract::getBinary(type, Abstract::DivU)) {
+          return binary->left;
+        }
       }
     }
+    // TODO: v128 not implemented yet
```

Comment:
why doesn't this optimization work for simd? seems like it could just work, if the Abstract stuff were hooked up - is that the issue?

Reply:
Yes, the actual problem is that the abstract stuff isn't hooked up yet. However, that will require a little bit more work because abstract will need to gain a notion of SIMD lanes, so I'd rather save that for a separate PR.

Reply:
Ok, lgtm.

Reply:
It's also probably fine to not do that optimization for now, if it's not easy to do - benefis are likely small.